### PR TITLE
fixes frontend / backend wiring for aggregate reports

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts [hunk 1]
```diff
@@ -532,6 +532,16 @@ export class AggregateReportTableComponent implements OnInit {
...
+      (a: AggregateReportItem, b: AggregateReportItem) => {
+        if (a.subgroup.id === 'Overall:') {
+          return -1;
+        }
+        if (b.subgroup.id === 'Overall:') {
+          return 1;
+        }
+        return 0;
+      },
       ordering(byString).on(({ subgroup }) => subgroup.id).compare
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: later we need a more permanent way to sort filtered subgroups vs basic subgroup rows

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts [hunk 2]
```diff
@@ -348,9 +348,31 @@ export class AggregateReportRequestMapper {
       subgroupFilters.migrantStatuses = subgroup.migrantStatusCodes;
     }
     if (notNullOrEmpty(subgroup.section504Codes)) {
-      subgroupFilters.section504Codes = subgroup.section504Codes;
+      subgroupFilters.section504s = subgroup.section504Codes;
     }
     return subgroupFilters;
   }
 
+  // TODO consolidate type names to remove need for extra mapping
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts]: hoping to remove this mapping layer in a followup

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts [hunk 3]
```diff
@@ -118,8 +118,8 @@ export class AggregateReportRequestMapper {
   toSettings(request: AggregateReportRequest, options: AggregateReportOptions): Observable<AggregateReportFormSettings> {
 
     const query: AggregateReportQuery = request.query;
-    const queryType: string = request.query.queryType || 'Basic';
-    const reportType: string = request.query.reportType || 'GeneralPopulation';
+    const queryType: string = request.query.subgroups ? 'FilteredSubgroup' : 'Basic';
+    const reportType: string = this.fromServerReportType(request.query.reportType);
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts]: @wtritch backend changed `queryType` to `reportType` will this affect prod reports?

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts (hunk 1 above)
Comment (agorina) [reply webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: +1

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts (hunk 3 above)
Comment (agorina) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts]: I believe that AggregateReportRequestDeserializer would take care of it, but would like Will to confirm.

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts [hunk 4]
```diff
@@ -348,9 +348,31 @@ export class AggregateReportRequestMapper {
       subgroupFilters.migrantStatuses = subgroup.migrantStatusCodes;
     }
     if (notNullOrEmpty(subgroup.section504Codes)) {
-      subgroupFilters.section504Codes = subgroup.section504Codes;
+      subgroupFilters.section504s = subgroup.section504Codes;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-request.mapper.ts]: fixed a bug while i was at this

**Pull request merged**